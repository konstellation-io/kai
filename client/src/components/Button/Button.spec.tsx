// import React from 'react';
// import { render, fireEvent, cleanup } from '@testing-library/react';
// import Button from './Button';

// afterEach(cleanup);

// it('Render Button without crashing', () => {
//   const { container } = render(<Button />);
//   expect(container).toMatchSnapshot();
// });

// it('Shows right texts', () => {
//   const { rerender, container } = render(<Button />);
//   expect(container.textContent).toBe('Button');

//   rerender(<Button label="New Text" />);
//   expect(container.textContent).toBe('New Text');

//   rerender(
//     <Button
//       label="Other Text"
//       type="dark"
//       onClick={function() {}}
//       primary
//       disabled
//     />
//   );
//   expect(container.textContent).toBe('Other Text');
// });

// it('Handles click events', () => {
//   const clickMock = jest.fn(() => true);

//   const { getByText } = render(<Button onClick={clickMock} />);
//   fireEvent.click(getByText('Button'));

//   expect(clickMock).toHaveBeenCalledTimes(1);
// });
