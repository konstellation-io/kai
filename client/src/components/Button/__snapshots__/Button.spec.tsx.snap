// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button without crashing 1`] = `
<div>
  <div
    class="btn label"
  >
    Button
  </div>
</div>
`;
